chaoscenter/graphql/server/pkg/chaos_experiment/ops/service.go

```diff
@@ -754,3 +899,163 @@ func (c *chaosExperimentService) UpdateRuntimeCronWorkflowConfiguration(cronWork
+			Operation:            probeInputs.K8sProbeInputs.Operation,
+			Attempt:              &probeInputs.RunProperties.Attempt,
+			Retry:                &probeInputs.RunProperties.Retry,
+			ProbePollingInterval: &probeInputs.RunProperties.ProbePollingInterval,
+			EvaluationTimeout:    &probeInputs.RunProperties.EvaluationTimeout,
+			StopOnFailure:        &probeInputs.RunProperties.StopOnFailure,
+			Group:                &probeInputs.K8sProbeInputs.Group,
+			ResourceNames:        &probeInputs.K8sProbeInputs.ResourceNames,
+			Namespace:            &probeInputs.K8sProbeInputs.Namespace,
+			FieldSelector:        &probeInputs.K8sProbeInputs.FieldSelector,
+			LabelSelector:        &probeInputs.K8sProbeInputs.LabelSelector,
+		}
+	case model.ProbeTypeCmdProbe:
+		source, _ := json.Marshal(probeInputs.CmdProbeInputs.Source)
+		sourcePtr := string(source)
+		kubernetesCMDProperties = &model.KubernetesCMDProbeRequest{
+			ProbeTimeout: probeInputs.RunProperties.ProbeTimeout,
+			Interval:     probeInputs.RunProperties.Interval,
+			Command:      probeInputs.CmdProbeInputs.Command,
+			Comparator: &model.ComparatorInput{
+				Type:     probeInputs.CmdProbeInputs.Comparator.Type,
+				Criteria: probeInputs.CmdProbeInputs.Comparator.Criteria,
+				Value:    probeInputs.CmdProbeInputs.Comparator.Value,
+			},
+			Attempt:              &probeInputs.RunProperties.Attempt,
+			Retry:                &probeInputs.RunProperties.Retry,
+			ProbePollingInterval: &probeInputs.RunProperties.ProbePollingInterval,
+			InitialDelay:         &probeInputs.RunProperties.InitialDelay,
+			StopOnFailure:        &probeInputs.RunProperties.StopOnFailure,
+			Source:               &sourcePtr,
+		}
+	}
+
+	return model.ProbeRequest{
+		Name:                     probeInputs.Name,
+		Type:                     model.ProbeType(probeInputs.Type),
+		K8sProperties:            k8sProperties,
+		KubernetesHTTPProperties: kubernetesHTTPProperties,
+		KubernetesCMDProperties:  kubernetesCMDProperties,
+		PromProperties:           promProperties,
+		InfrastructureType:       model.InfrastructureType(model.InfrastructureTypeKubernetes),
+		Tags:                     []string{},
+	}, nil
+}
+
+func insertProbeRefAnnotation(rawYaml, value string) (string, error) {
+	var data interface{}
+
+	err := yaml.Unmarshal([]byte(rawYaml), &data)
+	if err != nil {
+		return "", err
+	}
+
+	dataMap := data.(map[string]interface{})
+
+	metadata := dataMap["metadata"]
+	if metadata == nil {
+		return "", errors.New("metadata not found")
+	}
+
+	annotations := metadata.(map[string]interface{})["annotations"]
+	if annotations == nil {
+		// create new annotations
+		annotations = make(map[string]interface{})
+		metadata.(map[string]interface{})["annotations"] = annotations
+	}
+
+	annotations.(map[string]interface{})["probeRef"] = value
+
+	result, err := yaml.Marshal(dataMap)
+	if err != nil {
+		return "", err
+	}
+
+	return string(result), nil
+}
```

Comment:
These two functions can be moved to the probe utils. Also we have utils there to add all the types of probes, could that be reused here for the `ProbeInputsToProbeRequest` func?

Reply:
There's no similar function in utils, so I moved both functions to utils.